bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/gnirs/GnirsRule.java

```diff
@@ -43,6 +43,32 @@ public IConfigMatcher getMatcher() {
         }
     };
 
+    private static final IConfigRule SHORT_RED_CAMERA_RULE = new IConfigRule() {
+        private static final String MESSAGE = "The short red camera is currently unavailable.";
+
+        @Override
+        public Problem check(final Config config, final int step, final ObservationElements elems, final Object state) {
+            final Camera camera = (Camera) SequenceRule.getInstrumentItem(config, InstGNIRS.CAMERA_PROP);
+            if (camera == null) return null;
```

Comment:
My guess would have been that if the camera is `null`, _then_ we apply the pixel scale and wavelength check.  If not `null`, then the camera setting itself would prevail.  I don't know how it works though.

As is even if the camera is set explicitly to, say, short blue, we're doing the other property checks to make a determination.  Is this what they want?

Reply:
Good question. Can you se e.g. short blue and the pixel scale and wavelength so that short red is implied?
I guess in that case the explicit value should win?
Can the camera ever be null?

Reply:
I believe the camera setting is completely independent of the pixel scale and wavelength, in the `ocs` codebase at least.  I don't know how the hardware works.  It is usually `null` but can be explicitly set in engineering mode.  I would assume that when the camera is explicitly set, we use that and when not we use the properties but we should clarify.